src/libraries/System.Private.CoreLib/src/System/IO/Directory.cs

```diff
@@ -285,9 +286,15 @@ public static void Move(string sourceDirName, string destDirName)
 
             // Windows will throw if the source file/directory doesn't exist, we preemptively check
             // to make sure our cross platform behavior matches .NET Framework behavior.
-            if (!FileSystem.DirectoryExists(fullsourceDirName) && !FileSystem.FileExists(fullsourceDirName))
+            if (!FileSystem.DirectoryExists(fullsourceDirName) && !sourcePathIsFile)
                 throw new DirectoryNotFoundException(SR.Format(SR.IO_PathNotFound_Path, fullsourceDirName));
 
+            // To match API specification, we check if the source path is a file and ends with a trailing slash.
+            // While having a file as source is no problem for any OS, Windows will throw an error if the source path has a trailing slash.
+            // To match platform agnostic behaviour, we check that before handing over to OS specific implementation.
+            if (sourcePathIsFile && Path.EndsInDirectorySeparator(fullsourceDirName))
+                throw new IOException(SR.Format(SR.IO_PathNotFound_Path, fullsourceDirName));
```

Comment:
Is this case unit testable?

Reply:
Should already be present
https://github.com/dotnet/runtime/blob/206dba75f00e67c839709e0632b22819329d73d0/src/libraries/System.IO.FileSystem/tests/Directory/Move.cs#L96-L102